ls1intum/Artemis
# `Exam mode`: Fix an issue when loading modeling exercise results in the summary of test runs

### Checklist
#### General
#### Server

### Description
Fixes https://github.com/ls1intum/Artemis/issues/7561

### Steps for Testing
Prerequisites:
- 1 Instructor

1. Create an exam with a modeling exercise _(the release date of the grades must be in the future, the error does not occur once the results are released)_
2. Create an exam test run
3. participate in the test run
4. Assess the test run
5. Click on Test Run > Summary
6. See that no error messages are displayed and the exam results are loaded properly _(you will need to assess it manually if you want a grading to be displayed)_

### Testserver States
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress
#### Code Review
#### Manual Tests

### Test Coverage
#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [AuthorizationCheckService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5733/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service/AuthorizationCheckService.html) | 93% | ✅ |



### Screenshots
#### New
results loaded, no error messages

https://github.com/ls1intum/Artemis/assets/63976129/95b9b30f-d734-4299-bf3f-6e715f527959

#### Old
modeling exercise results are not loaded, error message displayed

https://github.com/ls1intum/Artemis/assets/63976129/31984b14-b0c6-43d4-b21e-0bd0e9f0c8ad
